node/blockAPI/baseBlock.go

```diff
@@ -15,94 +17,72 @@ type baseAPIBockProcessor struct {
 	store                    dataRetriever.StorageService
 	marshalizer              marshal.Marshalizer
 	uint64ByteSliceConverter typeConverters.Uint64ByteSliceConverter
-	historyProc              fullHistory.HistoryHandler
+	historyProc              fullHistory.HistoryRepository
 	unmarshalTx              func(txBytes []byte, txType string) (*transaction.ApiTransactionResult, error)
 }
 
-func (bap *baseAPIBockProcessor) getNormalTxsFromMB(miniblock *block.MiniBlock, epoch uint32) []*transaction.ApiTransactionResult {
-	txs := make([]*transaction.ApiTransactionResult, len(miniblock.TxHashes))
-	unit := dataRetriever.TransactionUnit
-	for idx := 0; idx < len(miniblock.TxHashes); idx++ {
-		txBytes, err := bap.getFromStorerWithEpoch(unit, miniblock.TxHashes[idx], epoch)
-		if err != nil {
-			continue
-		}
+var log = logger.GetOrCreate("node/blockAPI")
 
-		tx, err := bap.unmarshalTx(txBytes, "normal")
-		if err != nil {
-			continue
-		}
-
-		txs[idx] = tx
+func (bap *baseAPIBockProcessor) getTxsByMb(mbHeader *block.MiniBlockHeader, epoch uint32) []*transaction.ApiTransactionResult {
+	mbBytes, err := bap.getFromStorerWithEpoch(dataRetriever.MiniBlockUnit, mbHeader.Hash, epoch)
+	if err != nil {
+		log.Warn("cannot get miniblock from storage",
+			"hash", hex.EncodeToString(mbHeader.Hash),
+			"error", err.Error())
+		return nil
 	}
 
-	return txs
-}
-
-func (bap *baseAPIBockProcessor) getRewardTxsFromMB(miniblock *block.MiniBlock, epoch uint32) []*transaction.ApiTransactionResult {
-	txs := make([]*transaction.ApiTransactionResult, len(miniblock.TxHashes))
-	unit := dataRetriever.RewardTransactionUnit
-	for idx := 0; idx < len(miniblock.TxHashes); idx++ {
-		txBytes, err := bap.getFromStorerWithEpoch(unit, miniblock.TxHashes[idx], epoch)
-		if err != nil {
-			continue
-		}
-
-		tx, err := bap.unmarshalTx(txBytes, "rewardTx")
-		if err != nil {
-			continue
-		}
-
-		txs[idx] = tx
+	miniBlock := &block.MiniBlock{}
+	err = bap.marshalizer.Unmarshal(miniBlock, mbBytes)
+	if err != nil {
+		log.Warn("cannot unmarshal miniblock",
+			"hash", hex.EncodeToString(mbHeader.Hash),
+			"error", err.Error())
+		return nil
 	}
 
-	return txs
+	switch miniBlock.Type {
+	case block.TxBlock:
+		return bap.getTxsFromMiniblock(miniBlock, epoch, "normal", dataRetriever.TransactionUnit)
+	case block.RewardsBlock:
+		return bap.getTxsFromMiniblock(miniBlock, epoch, "rewardTx", dataRetriever.RewardTransactionUnit)
```

Comment:
Could be `reward` instead of `rewardTx`.

Reply:
done.